src/group_theory/p_group.lean

```diff
@@ -248,6 +248,18 @@ lemma to_sup_of_normal_left' {H K : subgroup G} (hH : is_p_group p H) (hK : is_p
   (hHK : K ≤ H.normalizer) : is_p_group p (H ⊔ K : subgroup G) :=
 (congr_arg (λ H : subgroup G, is_p_group p H) sup_comm).mp (to_sup_of_normal_right' hK hH hHK)
 
+/-- finite p-groups with different p have coprime orders -/
+lemma coprime_card_of_ne {G₂ : Type*} [group G₂]
+  (p₁ p₂ : ℕ) [hp₁ : fact p₁.prime] [hp₂ : fact p₂.prime] (hne : p₁ ≠ p₂)
+  (H₁ : subgroup G) (H₂ : subgroup G₂) [fintype H₁] [fintype H₂]
+  (hH₁ : is_p_group p₁ H₁) (hH₂ : is_p_group p₂ H₂) :
```

Comment:
Not very important, but does it make sense to add `hne` as last argument? Intuitively I expect that this is most likely the thing that one wants to turn into the new goal.
I think I would expect the order `H₁ H₂ p₁ p₂ hH₁ hH₂ hne`.

Reply:
Hmm, not sure. Where I use this (in https://github.com/leanprover-community/mathlib/pull/11778/files) there are lots of `intro p1 p2 hne` and `apply foo p1 p2 hne` idioms for different lemmas that all start with “given two distinct primes”. Moving the `hne` away from the `p`’s would – in that style at least – be more annoying than helpful.

Reply:
Fair enough. I'm certainly fine with keeping it as it is.